core/src/main/java/org/apache/accumulo/core/util/MapCounterInt.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Set;
+
+/**
+ * A version of MapCounter for counting with Integers
+ */
+public class MapCounterInt<KT> {
+
+  static class MutableInt {
+    int i = 0;
+  }
+
+  private HashMap<KT,MutableInt> map;
+
+  public MapCounterInt() {
+    map = new HashMap<>();
+  }
+
+  public int increment(KT key, int i) {
+    MutableInt mutableInt = map.get(key);
+    if (mutableInt == null) {
+      mutableInt = new MutableInt();
+      map.put(key, mutableInt);
+    }
+
+    mutableInt.i += i;
+
+    if (mutableInt.i == 0) {
+      map.remove(key);
+    }
+
+    return mutableInt.i;
+  }
+
+  public int decrement(KT key, int i) {
+    return increment(key, -1 * i);
+  }
+
+  public int get(KT key) {
+    MutableInt mutableInt = map.get(key);
+    if (mutableInt == null) {
+      return 0;
+    }
+
+    return mutableInt.i;
+  }
+
+  public Set<KT> keySet() {
+    return map.keySet();
+  }
+
+  public Collection<Integer> values() {
+    Collection<MutableInt> vals = map.values();
```

Comment:
This method could be update to return `Collections2.transform(map.values(), mi->mi.i)`

Reply:
Looking at the javadoc for that Guava transform I noticed this...
"When a live view is <i>not</i> needed, it may be faster to copy the transformed collection and use the copy."

Which makes me wonder if what we have is faster.  Or is that just saying to make a copy of what is returned?

Reply:
I would stick with what is there after reading that comment.